pkg/services/ngalert/notifier/channels/testing.go

```diff
@@ -29,63 +26,79 @@ func (f *fakeImageStore) GetImage(_ context.Context, token string) (*ngmodels.Im
 	return nil, ngmodels.ErrImageNotFound
 }
 
-// newFakeImageStore returns an image store with a test image.
-// The image has a token and a URL, but does not have a file on disk.
-func newFakeImageStore() ImageStore {
-	return &fakeImageStore{
-		Images: []*ngmodels.Image{
-			{
-				Token:     "test-image",
-				URL:       "https://www.example.com/test-image.jpg",
-				CreatedAt: time.Now().UTC(),
-			},
-		},
+// newFakeImageStore returns an image store with N test images.
+// Each image has a token and a URL, but does not have a file on disk.
+func newFakeImageStore(n int) ImageStore {
+	s := fakeImageStore{}
+	for i := 0; i < n; i++ {
+		s.Images = append(s.Images, &ngmodels.Image{
+			Token:     fmt.Sprintf("test-image-%d", i),
+			URL:       fmt.Sprintf("https://www.example.com/test-image-%d.jpg", 1),
+			CreatedAt: time.Now().UTC(),
+		})
 	}
+	return &s
 }
 
-// newFakeImageStoreWithFile returns an image store with a test image.
-// The image has a token, path and a URL, where the path is 1x1 transparent
-// PNG on disk. The test should call deleteFunc to delete the image from disk
+// newFakeImageStoreWithFile returns an image store with N test images.
+// Each image has a token, path and a URL, where the path is 1x1 transparent
+// PNG on disk. The test should call deleteFunc to delete the images from disk
 // at the end of the test.
 // nolint:deadcode,unused
-func newFakeImageStoreWithFile(t *testing.T) (ImageStore, deleteFunc) {
+func newFakeImageStoreWithFile(t *testing.T, n int) ImageStore {
+	var (
+		files []string
+		s     fakeImageStore
+	)
+
+	t.Cleanup(func() {
+		// remove all files from disk
+		for _, f := range files {
+			if err := os.Remove(f); err != nil {
+				t.Logf("failed to delete file: %s", err)
+			}
+		}
+	})
+
+	for i := 0; i < n; i++ {
+		file, err := newTestImage()
+		if err != nil {
+			t.Fatalf("failed to create test image: %s", err)
+		}
+		files = append(files, file)
+		s.Images = append(s.Images, &ngmodels.Image{
+			Token:     fmt.Sprintf("test-image-%d", i),
+			Path:      file,
+			URL:       fmt.Sprintf("https://www.example.com/test-image-%d", i),
+			CreatedAt: time.Now().UTC(),
+		})
+	}
+
+	return &s
+}
+
+// nolint:deadcode,unused
```

Comment:
I'm not sure if this is still necessary here.

Reply:
That's what I thought too, but the linter didn't like it because the function that calls this function is itself unused.